### PR TITLE
[PLAT-11289] Add createNavigationContainer method

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Add `createNavigationContainer` method to wrap the react navigation container

## Changeset

- Added method for wrapping provided component and monkey patching `onStateChange` method.
- Updated jest configuration for component testing

## Testing

Unit tests for route changes

### CODE REVIEW

**Comment by gingerbenw** [comment]
not sure why the test coverage changes aren't showing anything... 🤔

File: packages/platforms/react-native/tests/retry-queue/index.test.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ import FileSystemFake from '../utilities/file-system-fake'
 import { InMemoryDelivery, makePayloadCreator } from '@bugsnag/js-performance-test-utilities'
 import { AppState } from 'react-native'
 
-// eslint-disable-next-line jest/no-mocks-import
```
Comment (gingerbenw) [inline packages/platforms/react-native/tests/retry-queue/index.test.ts:7]: oh yeah, I fixed this somehow!

Comment (gingerbenw) [reply packages/platforms/react-native/tests/retry-queue/index.test.ts:7]: or did I un-fix it?

File: packages/react-navigation/lib/create-navigation-container.tsx
Comment (gingerbenw) [inline packages/react-navigation/lib/create-navigation-container.tsx:1]: this is actually the main new feature

File: packages/react-navigation/lib/create-navigation-container.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import BugsnagPerformance from "@bugsnag/react-native-performance";
+import { NavigationContainer, NavigationContainerProps, NavigationContainerRef, useNavigationContainerRef } from "@react-navigation/native";
+import React, { forwardRef, useRef } from 'react';
+import { NavigationContextProvider } from "./navigation-context";
+
+export const createNavigationContainer = <P extends {}>(Container = NavigationContainer) =>
+  forwardRef<NavigationContainerRef<P>, NavigationContainerProps>(
+    (props, _ref) => {
```
Comment (yousif-bugsnag) [inline packages/react-navigation/lib/create-navigation-container.tsx:8]: Not sure if this `forwardRef` is required? the passed in `_ref` isn't used anywhere

Comment (gingerbenw) [reply packages/react-navigation/lib/create-navigation-container.tsx:8]: We're using `forwardRef` here to access the props of the component provided as a parameter to the function, but I don't think we actually need the ref, as we will generally just be calling this function with the default `NavigationContainer` exported from `@react-navigation/native` so shouldn't have any imperative handling with `refs` before we access it.

**Review by yousif-bugsnag — APPROVED** [review]

**Pull request merged**